imagebuildah/executor.go

```diff
@@ -776,6 +843,17 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
 		return "", nil, err
 	}
 	logrus.Debugf("printing final image id %q", imageID)
+	if b.buildOutput != "" {
+		logrus.Debugf("Generating custom build output with options %q", b.buildOutput)
+		path, isDir, isStdout, err := internalParse.GetBuildOutput(b.buildOutput)
```

Comment:
Wouldn’t it be better to parse all of this (and report errors) before starting to actually build anything?

(And then, the result of `GetBuildOutput` should probably be an object that can just be passed on as a single unit, not a set of loose independent variables the caller needs to manage one at a time.)

Reply:
This is a nice idea, added in latest commit.